### DISCUSSION TITLE
stateful material property for an IC object

### DISCUSSION THREAD

**Discussion by steam-simulator** [idaholab/moose#24242] [Q&A General] [answered]
Hello @GiudGiud, can I describe a stateful material property in an IC object? I’m getting the following 
error: `'declareProperty' was not declared in this scope.`

I have the following in my header and src:

```
MaterialProperty<Real> & _b;
const MaterialProperty<Real> & _b_old;   
```
 
```
_b(declareProperty<Real>("initial_frac")),
_b_old(getMaterialPropertyOld<Real>("initial_frac_old")),
```
Thanks.

**Comment by GiudGiud** [comment, marked as answer]
Hello

Initial conditions cannot declare material properties.
Only a Material object can
https://mooseframework.inl.gov/docs/doxygen/moose/classMaterialBase.html#ac0bb20e3b6934feb9bbe61da109053fc

Guillaume

**Reply by steam-simulator** [reply]
Yeah, I thought so. Thanks.
